python/paddle/fluid/dataset.py

```diff
@@ -277,6 +277,73 @@ def set_use_var(self, var_list):
                     "Currently, fluid.dataset only supports dtype=float32, dtype=int32 and dtype=int64"
                 )
 
+    def check_use_var_with_data_generator(self, var_list, data_generator_class,
+                                          test_file):
+        """
+         Var consistency insepection of use_var_list and data_generator data.
+
+        Examples:
+            .. code-block:: python
+
+              # required: skiptest
+              import paddle.fluid as fluid
+              from dataset_generator import CTRDataset
+              dataset = fluid.DatasetFactory().create_dataset()
+              generator_class = CTRDataset()
+              dataset.check_use_var_with_data_generator([data, label], generator_class, "data/part-00000")
+
+        Args:
+            var_list(list): variable list
+            data_generator_class(class): data_generator class
+            test_file(str): local test file path
+        """
+
+        f = open(test_file, "r")
+        var_len = len(var_list)
+
+        while True:
+            line = f.readline()
+            if line:
+                line_iter = data_generator_class.generate_sample(line)
+                for user_parsed_line in line_iter():
+                    data_gen_len = len(user_parsed_line)
+                    if var_len != data_gen_len:
+                        raise ValueError(
+                            "var length mismatch error: var_list = %s vs data_generator = %s"
+                            % (var_len, data_gen_len))
+
+                    for i, ele in enumerate(user_parsed_line):
+                        # print(ele[0], ele[1])
+
+                        if len(ele[1]) == 0:
+                            raise ValueError(
+                                "var length error: var %s's length in data_generator is 0"
+                                % ele[0])
+
+                        if var_list[
+                                i].dtype == core.VarDesc.VarType.FP32 and not all(
+                                    isinstance(ele, float) for ele in ele[1]):
```

Comment:
1. write var_list[i] in one line?  2. if not all(isinstance(ele,float)),  you'd better pring ele out. otherwise, user doesn't know which element in the current line

Reply:
1. pre-commit format in this way...
2. have already printed ele out, see "raise TypeError"